tests/unit/lms/views/lti/deep_linking_test.py

```diff
@@ -211,6 +211,10 @@ def test_it_for_v11(
                 {"group_set": "1", "title": "title"},
                 {"group_set": "1", "title": "title"},
             ),
+            (
+                {"auto_grading_config": {"key": "value"}},
+                {"auto_grading_config": '{"key": "value"}'},
```

Comment:
Are we not validating that this fulfills `_AutoGradingConfig` definition? 🤔 

I would expect these values to be invalid.

Reply:
Yes, good point but I'm afraid I don't a satisfactory answer.

The method here is unaware of the types, it just passes along what it receives, and in the case of these settings it serlizes it as json. That's what the test does. I included the private name of the method in the test name (one extra `_`)

The marshmallow schema is validated before we get into the the code of individual views and it gets serialized into `request.parsed_params`, if that not valid the view is never called.

That being said I added a very broad valid/invalid tests for the schema itself :+1:

Reply:
> The marshmallow schema is validated before we get into the the code of individual views and it gets serialized into `request.parsed_params`, if that not valid the view is never called.

Oh, I see 👍🏼